Build failure on Apple Silicon

As I mentioned in #242, I'm trying to build luvi on Apple Silicon. However, this fails with a linker error:
```
ld: missing LC_SEGMENT file 'jitted_tmp/src/lua/init.lua_luvi_generated.o' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [luvi] Error 1
make[1]: *** [CMakeFiles/luvi.dir/all] Error 2
make: *** [all] Error 2
```
CI logs available at https://github.com/Homebrew/homebrew-core/pull/69036/checks?check_run_id=1739201673.

The error starts [here](https://github.com/Homebrew/homebrew-core/pull/69036/checks?check_run_id=1739201673#step:7:338) and the CMake invocation is [here](https://github.com/Homebrew/homebrew-core/pull/69036/checks?check_run_id=1739201673#step:7:155).

This compiles successfully on the Intel CI nodes. As previously mentioned, I'm building with a custom libuv (v1.40), a custom LuaJIT (OpenResty v2.1-20201229), and a shared libluv (v1.36.0, built with the aforementioned libuv and LuaJIT).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,10 +170,6 @@ lua_add_executable(luvi
   ${lpeg_re_lua}
 )
 
-if(APPLE)
-  set(CMAKE_EXE_LINKER_FLAGS "-pagezero_size 10000 -image_base 100000000 ${CMAKE_EXE_LINKER_FLAGS}")
-endif()
-
 if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
   set(CMAKE_EXE_LINKER_FLAGS "-Wl,-E")
 endif()

